src/bytes.rs

```diff
@@ -1666,7 +1666,7 @@ impl<'a> From<&'a [u8]> for BytesMut {
             BytesMut::new()
         } else if len <= INLINE_CAP {
             unsafe {
-                let mut inner: Inner = mem::uninitialized();
+                let mut inner: Inner = mem::MaybeUninit::uninit().assume_init();
```

Comment:
This is not the correct usage of `MaybeUninit`.

see https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#examples

Reply:
it is fine though, since it is equivalent of original code

Reply:
This is not fine, otherwise `uninitialized` will not be deprecated.

Reply:
it is deprecated because it was easy to misuse and some edge cases, there it is used just fine